src/Client.php

```diff
@@ -100,22 +101,84 @@ private function getSession(?string $alias = null): SessionInterface
         return $this->boundSessions[$alias] = $this->startSession($alias, $this->defaultSessionConfiguration);
     }
 
+    /**
+     * Executes an operation with automatic retry on alternative drivers when connection exceptions occur.
+     *
+     * @template T
+     *
+     * @param callable(SessionInterface): T $operation The operation to execute
+     * @param string|null                   $alias     The driver alias to use
+     *
+     * @throws ConnectionPoolException When all available drivers have been exhausted
+     *
+     * @return T The result of the operation
+     */
+    private function executeWithRetry(callable $operation, ?string $alias = null)
+    {
+        $alias ??= $this->driverSetups->getDefaultAlias();
+        $attemptedDrivers = [];
+        $lastException = null;
+
+        while (true) {
+            try {
+                $driver = $this->driverSetups->getDriver($this->defaultSessionConfiguration, $alias);
```

Comment:
The idea here is sound, but the execution will have a bug. The driver setups will return the same driver every time. We will need to adjust the DriverSetupManager as well. Maybe we can add a `reset` method to the class for a specific alias, so it verifies the connections again and finds a new driver with a verified connection? If we go that route, we don't need any complex retry logic; catch the exception, reset the drivers in the driver setup manager, and try again.